### PR TITLE
Pools several unpooled overlay/temp objects

### LINKED ISSUE #18869: Medical Hologram made by pen lights not visible.
If you click on a mob while you have a pen light in your hands, it used to create a sound and a medical hologram. For a while now, the graphical part is missing, I only hear the sound but do not see the sprite.

That thing was quite useful to get the attention of half-dead people walking away from my healing hands.

### PR DESCRIPTION
**Pull request by ChangelingRain**
KOOOOOOOOOR but also PHIIIIIIIIIL

this saves like, a tiny amount but probably quite a bit for dragons because they weren't pooling the fireballs.

Fixes #18869

### CODE REVIEW

File: code/game/objects/items/devices/flashlight.dm [hunk 1]
```diff
@@ -127,7 +127,7 @@
 	icon_state = "medi_holo"
 	duration = 30
 
-/obj/effect/medical_holosign/New(loc, creator)
+/obj/effect/overlay/temp/medical_holosign/New(loc, creator)
```
Comment (swankcookie) [inline code/game/objects/items/devices/flashlight.dm]: lines 130-134 are unnecessary anyway if you just use my fix. why change it?

**Comment by Fox-McCloud** [comment]
Likely to result in *less* overall performance.

Even with the numerous changes and updates to pooling, it still results in a net loss of performance, rather than gains.

@MrStonedOne will even concur with this one, unless things have recently changed.

**Comment by ChangelingRain** [comment]
Someone told me I should be pooling these, so I'm pooling these.

**Comment by Fox-McCloud** [comment]
Well, I mean, if you want consistency, then sure, but...otherwise, you're giving yourself less CPU to work with by doing this.

**Comment by MrStonedOne** [comment]
the datum var improve will add a slight increase in new overhead and a slight decrease in var access overhead, so we shall see what 511 brings

**Pull request merged**